### PR TITLE
Make SuperTextField handle DONE action (Resolves #661)

### LINKED ISSUE #661: [SuperTextField] - on iOS web, pressing "DONE" on the keyboard doesn't do anything
When entering text in a `SuperTextField` on iOS web, at the top of the keyboard is a button that reads "DONE".

Pressing "DONE" has no effect.

I would expect pressing "DONE" to close the keyboard and remove focus from the field in question.

### PR DESCRIPTION
**Pull request by angelosilvestre**
Make SuperTextField handle DONE action. Resolves https://github.com/superlistapp/super_editor/issues/661

Here I'm not sure if we should unfocus even if the widget has a `onPerformActionPressed` defined. I could change this if needed. 

```dart
  void _onPerformActionPressed(TextInputAction action) {
    if (action == TextInputAction.done) {
      _focusNode.unfocus();
    }
    widget.onPerformActionPressed?.call(action);
  }
```

Maybe we should expose `onPerformActionPressed` on `SuperTextField`'s public API to write a test to ensure it's beeing called.

### CODE REVIEW

**Comment by matthew-carroll** [comment]
I looked through some of the existing API structure related to this and I think we want to make a few changes in this PR.

1. I think we should remove the `performAction` property in the Android and iOS text fields. Instead, the client should pass a controller that's configured with a perform action callback.
2. In the mobile text fields, I think we only want to take an action if the controller doesn't have a callback set on it.
3. When we handle the action, I think we need to implement all the reasonable defaults. Take a look at Flutter's enum and implement any action where we can reasonably do the right thing: https://api.flutter.dev/flutter/services/TextInputAction.html
4. Add a test for every action that we support.
5. Add a test that ensures the default action only happens when the controller has a `null` callback, and that a non-`null` callback is called.

In the future we'll figure out what we need to expose in `SuperTextField`, but it's not clear to me what that should be. In fact, we'll probably need to take another look at the concept of `ImeAttributedTextController`. I'm not sure it's the right tool to begin with.

**Comment by angelosilvestre** [comment]
Should we handle `go`, `search` and `send`  unfocusing the field like in `done` ?

I've made a change that is unrelated to the original issue:

```dat
bool get _isMultiline => widget.minLines != 1 || widget.maxLines != 1;
```
`widget.minLines` accepts null, so without setting `minLines` the field was beeing considered multiline

 `widget.maxLines` also accepts null, but it defaults to 1 in the constructor

File: super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart [hunk 1]
```diff
@@ -289,7 +290,11 @@ class ImeAttributedTextEditingController extends AttributedTextEditingController
 
   @override
   void performAction(TextInputAction action) {
-    _onPerformActionPressed?.call(action);
+    if (_realController is ImeAttributedTextEditingController && (_realController as ImeAttributedTextEditingController).onPerformActionPressed != null) {
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart:293]: Do we need this if-statement? If our inner controller is also an `ImeAttributedTextEditingController` then this same situation could happen inside that controller, too. I think we're better off assuming that the client controller isn't an `ImeAttributedTextEditingController`.

File: super_editor/test/super_textfield/super_textfield_input_actions_test.dart [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+
+      // Simulate a tap at the action button
+      await tester.testTextInput.receiveAction(TextInputAction.previous);
+      await tester.pump();
+
+      // Ensure focus has moved to next field
+      expect(focusNodeFirstField.hasFocus, true);
+    });
+
+    group('having a textController with onPerformActionPressed set', () {
```
Comment (matthew-carroll) [inline super_editor/test/super_textfield/super_textfield_input_actions_test.dart:83]: "having a textController with onPerformActionPressed set" -> "with custom onPerformActionPressed callback"

File: super_editor/lib/src/infrastructure/super_textfield/ios/ios_textfield.dart [hunk 3]
```diff
@@ -356,6 +356,22 @@ class SuperIOSTextFieldState extends State<SuperIOSTextField>
...
+      case TextInputAction.done:
+        _focusNode.unfocus();
+        break;
+      case TextInputAction.next:
+        _focusNode.nextFocus();
+        break;
+      case TextInputAction.previous:
+        _focusNode.previousFocus();
+        break;
+      default:
```
Comment (matthew-carroll) [inline super_editor/lib/src/infrastructure/super_textfield/ios/ios_textfield.dart:371]: For your question about the remaining actions: for now, let's log a warning that the user pressed an action button that doesn't have a default behavior

File: super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart (hunk 1 above)
Comment (angelosilvestre) [reply super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart:293]: `onPerformActionPressed` is defined only in `ImeAttributedTextEditingController` and `_realController` is defined as `AttributedTextEditingController`.

Comment (matthew-carroll) [reply super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart:293]: Right, but this class with `performAction()` **IS** an `ImeAttributedTextEditingController`, right? So this class has the callback that we want to call, just like the original implementation. Am I missing something?

Comment (angelosilvestre) [reply super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart:293]: When we pass an `textController` to `SuperTextField` it wraps it into another `ImeAttributedTextEditingController`: 

https://github.com/superlistapp/super_editor/blob/64e029e11660f3d5901d27344eba1105850c6bb4/super_editor/lib/src/infrastructure/super_textfield/super_textfield.dart#L154-L161

So, to call the original callback we need this.

Comment (matthew-carroll) [reply super_editor/lib/src/infrastructure/super_textfield/input_method_engine/_ime_text_editing_controller.dart:293]: Ok, let's change the text fields to only wrap the given controller if the given controller is not an `ImeAttributedTextEditingController`. Let me know if that creates any problems.

**Review by matthew-carroll — APPROVED** [review]
LGTM - Thanks.
